### ISSUE TITLE
[feature request] Support border anchors for `content()` elements

### ISSUE THREAD

**Issue by tapyu** [cetz-package/cetz#603]
In

```typ

#import "@preview/cetz:0.2.2": canvas, draw
#import draw: *

#let i = 50
#canvas({
  content(
    (),
    [hello world],
    anchor: "west",
    name: "hello",
  )
  line(
    "hello.south",
    (rel : (y: -2)),
  )
})
```

We are able to use compass border anchors (“north”, “north-east”, etc), but we cannot anchor by passing an angles (e.g. `"hello.30deg"`). This would be a very helpful feature for more complex Cetz figures.

**Comment by johannes-wolf** [comment]
Yes, this should be implemented. Since anchors of content do not get transformed by the current transformation, this is a bit more complicated, though.

**Comment by johannes-wolf** [comment]
What makes things really complicated is that `content` stays untransformed.
Maybe we should propagate adding something like `transform(<matrix>)` into Typst, allowing to set any transformation matrix for native Typst elements. This would then allow transforming content correctly with cetz current transformation.
